Issue2170 cdl sampling

mwetter opened the pull request:
This closes #2170

mwetter commented:
@milicag : Can you please review this PR. This change also removes state events that were generated with the previous implementation.

milicag commented on Buildings/Controls/OBC/CDL/Logical/Sources/SampleTrigger.mo:
```diff
@@ -2,13 +2,22 @@ within Buildings.Controls.OBC.CDL.Logical.Sources;
 block SampleTrigger "Generate sample trigger signal"
   parameter Modelica.SIunits.Time period(
     final min=Constants.small) "Sample period";
```
Should `Buildings/Controls/OBC/CDL/Integers/Sources/TimeTable.mo` line 11 also be `Constants.small`?

milicag commented on Buildings/Controls/OBC/CDL/Logical/Sources/SampleTrigger.mo:
```diff
@@ -53,18 +62,26 @@ equation
 </p>
-
+<p>
+The trigger signal is generated an infinite number of times, and aligned with time <code>time=delay</code>.
```
Maybe a small edit like this would help: `...and translated in time such that a trigger aligns with time <code>time=delay</code>`

milicag commented on Buildings/Controls/OBC/CDL/Logical/TriggeredTrapezoid.mo:
```diff
@@ -73,86 +73,7 @@ equation
           lineColor={0,0,255},
           textString="%name")}),
     Diagram(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},{
-            100,100}}), graphics={Line(points={{-80,-20},{-60,-20},{-30,40},{
```
Does this edit need a revision note?

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueFalseHold.mo:
```diff
@@ -2,27 +2,27 @@ within Buildings.Controls.OBC.CDL.Logical.Validation;
 
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul(
-      startTime=0, period=1500) "Boolean pulse input signal"
+      delay=0, period=1500) "Boolean pulse input signal"
```
new line please.

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueFalseHold.mo:
```diff
@@ -2,27 +2,27 @@ within Buildings.Controls.OBC.CDL.Logical.Validation;
     annotation (Placement(transformation(extent={{-40,20},{-20,40}})));
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul2(
-      startTime=0, period=1500) "Boolean pulse input signal"
+      delay=0, period=1500) "Boolean pulse input signal"
```
Here as well, new line.

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueFalseHold.mo:
```diff
@@ -36,27 +36,27 @@ model TrueFalseHold "Validation model for the TrueFalseHold block"
     annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul4(
-      startTime=100, period=1500) "Boolean pulse input signal"
+      delay=100, period=1500) "Boolean pulse input signal"
```
New line.

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueFalseHold.mo:
```diff
@@ -36,27 +36,27 @@ model TrueFalseHold "Validation model for the TrueFalseHold block"
     annotation (Placement(transformation(extent={{120,20},{140,40}})));
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul6(
-      startTime=100, period=1500) "Boolean pulse input signal"
+      delay=100, period=1500) "Boolean pulse input signal"
```
new line

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueHoldWithReset.mo:
```diff
@@ -2,7 +2,7 @@ within Buildings.Controls.OBC.CDL.Logical.Validation;
 
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul(
-      period=9000, startTime=300) "Boolean pulse input signal"
+      period=9000, delay=300) "Boolean pulse input signal"
```
new line

milicag commented on Buildings/Controls/OBC/CDL/Logical/Validation/TrueHoldWithReset.mo:
```diff
@@ -16,7 +16,7 @@ model TrueHoldWithReset "Validation model for the TrueHoldWithReset block"
     annotation (Placement(transformation(extent={{20,20},{40,40}})));
   Buildings.Controls.OBC.CDL.Logical.Sources.Pulse booPul2(
-      period=9000, startTime=300) "Boolean pulse input signal"
+      period=9000, delay=300) "Boolean pulse input signal"
```
new line

milicag commented on Buildings/package.mo:
```diff
@@ -460,23 +460,38 @@ have been <b style=\"color:blue\">improved</b> in a
+    </td>
+    <td valign=\"top\">Renamed parameter <code>startTime</code> to <code>delay</code>.<br/>
+                       This is for <a href=\"https://github.com/lbl-srg/modelica-buildings/issues/2170\">#2170</a>.<br/>
+                       For Dymola, a conversion script makes this change.
```
Do we have a pointer to the location of Dymola conversion scripts somewhere at the beginning of these revision notes?

mwetter replied:
A pointer is not needed because the conversion script is automatically involved per the Modelica Language Definition.

milicag commented on Buildings/package.mo:
```diff
@@ -460,23 +460,38 @@ have been <b style=\"color:blue\">improved</b> in a
+    <td valign=\"top\">Renamed parameter <code>startTime</code> to <code>delay</code>.<br/>
+                       This is for <a href=\"https://github.com/lbl-srg/modelica-buildings/issues/2170\">#2170</a>.<br/>
+                       For Dymola, a conversion script makes this change.
+                       The pulses will start at an integer multiple of the period of the pulse.</td>
```
Any, so: `The pulses will start at any integer multiple of the period of the pulse`

milicag reviewed (changes requested):
@mwetter Please address these minor comments as you find needed. I will look at the validation tests tomorrow morning.

mwetter commented:
@milicag : Thanks, I addressed the comments and will merge either https://github.com/lbl-srg/modelica-buildings/pull/2188/commits/4745346c53bf12be491e2e9978c59a54255c803f or https://github.com/lbl-srg/modelica-buildings/pull/2188/commits/9b95dfa669ab90efc27af7727407e74cb042655f (it it finishes soon) so I can use it for tomorrow's presentation.

milicag commented:
@mwetter I took a look at the validation tests and they look fine. One minor add that I can see being useful is an integer pulse validation case where at time 0 the pulse is at its `false` state integer value. Though I did observe this working fine, so it is optional.
